tests/admin_checks/tests.py

```diff
@@ -105,7 +111,30 @@ class CustomAdminSite(admin.AdminSite):
             self.assertEqual(errors, expected)
         finally:
             custom_site.unregister(Song)
-            admin.sites.system_check_errors = []
+
+    @override_settings(DEBUG=True)
+    def test_allows_checks_relying_on_other_modeladmins(self):
+        class MyBookAdmin(admin.ModelAdmin):
+            def check(self, **kwargs):
+                errors = super(MyBookAdmin, self).check(**kwargs)
+                author_admin = self.admin_site._registry.get(Author)
+                if author_admin is None:
+                    errors.append('AuthorAdmin missing!')
+                return errors
+
+        class MyAuthorAdmin(admin.ModelAdmin):
+            pass
+
+        admin.site.register(Book, MyBookAdmin)
+        admin.site.register(Author, MyAuthorAdmin)
+
+        try:
+            errors = checks.run_checks()
```

Comment:
The lines below do not belong into the try block, since they don't raise the exception.
```python
try:
    errors = checks.run_checks()
else:
    expected = []
    self.assertEqual(errors, expected)
finally:
    admin.site.unregister(Book)
    admin.site.unregister(Author)
```